src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1245,8 +1245,6 @@ private void WriteFrameworkReferences()
                         string itemSpec = _packageResolver.ResolvePackageAssetPath(library, asset.Path);
                         WriteItem(itemSpec, library);
                         WriteMetadata(MetadataKeys.PathInPackage, asset.Path);
-                        WriteMetadata(MetadataKeys.PackageName, library.Name);
```

Comment:
If we're not writing the PackageName metadata, should we be writing NuGetPackageId instead?  Also, should we be writing the version?

Reply:
I believe this happens in WriteItem(itemspec, library)

Reply:
Something came along here and added PackageName and PackageVersion when we already had NuGetPackageId, NuGetPackageVersion.

Reply:
That was my understanding and the reason I didn't add another write to NugetPackageId here. 
I also didn't remove PackageName from MetadataKeys because, based on what @nguerrera tagged as the commit where the redundancy arose, it seemed to me like the metadata wasn't redundant everywhere, just in this specific case. Did I misunderstand?

Reply:
If possible I think we should standardize on one metadata key or the other.

Reply:
I'm getting a lot of test failures when I try to standardize one way or the other.  Are NuGetPackageId and PackageName always the same thing? I'm also not clear as to what the difference is between the metadata in [MetadataNames](https://github.com/sfoslund/sdk/blob/RemoveExtraMetadata/src/Tasks/Common/ConflictResolution/MetadataNames.cs#L14) and [MetadataKeys](https://github.com/sfoslund/sdk/blob/RemoveExtraMetadata/src/Tasks/Common/MetadataKeys.cs#L65)- they both have NuGetPackageId.

Reply:
The duplication of MetadataNames and MetadataKeys looks like a historical missed opportunity to consolidate when code for conflict resolution that had been in another project was merged into this code base. We really only need one of these and can union the values together. That needn't be part of this change, though.

I am not aware of a scenario where PackageName and NuGetPackageId would be different. Could you share a branch where you attempted to standardize and an example test failure?

Reply:
I'm experimenting in [this branch](https://github.com/sfoslund/sdk/tree/RedundantMetadataExperiment) and the following tests are failing:
- Microsoft.NET.Build.Tests.GivenThatWeWantToBuildALibrary.It_marks_package_references_as_externally_resolved with input true and null (false is passing)
- Microsoft.NET.Build.Tests.GivenThatWeWantToResolveConflicts.The_same_references_are_used_with_or_without_DisableDefaultPackageConflictOverrides with input "netstandard2.0"

Reply:
Fixed the issue. I've consolidated to PackageName and removed NuGetPackageId from MetadataKeys.

Reply:
Hmm, I personally think NuGetPackageId was the better name, but more importantly, we should think about how breaking this will be. When I filed the issue, I think it was before we shipped with the new redundancy in this particular place. I hadn't sized up how breaking it would be elsewhere to consolidate.

I believe I've personally shared using NuGetPackageId on the items from ResolvePackageAssets here with customers. I also think that the non-SDK nuget build targets also use NuGetPackageId. And finally, for Content items that NuGet puts directly in generated props. it uses NuGetPackageId. So for these items, I think it would be better to stick to NuGetPackageId, but I'm less familiar with the items that have only PackageName today and how often that is used in code.

I tried looking through GitHub, but there's a lot of noise:

https://github.com/search?l=&p=1&q=NuGetPackageId+extension%3Atargets+extension%3Aprops+extension%3Acsproj+extension%3Avbproj&ref=advsearch&type=Code&utf8=%E2%9C%93

https://github.com/search?l=&p=1&q=PackageName+extension%3Atargets+extension%3Aprops+extension%3Acsproj+extension%3Avbproj&ref=advsearch&type=Code&utf8=%E2%9C%93

Reply:
Switched back to NuGetPackageId. It's hard to tell how often PackageName is used based on that search.